### ISSUE TITLE
how to use self-host agent for azure pipeline

### ISSUE THREAD

**Issue by tjinjin95** [microsoft/azure-pipelines-agent#3293]
Hi, I want create a self-host agent and use it in pipeline. however, I had some problems importing the image.
the procedures are list in follow.

1.  create a self-host agent according to the link : [https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/v2-windows?view=azure-devops]
![image](https://user-images.githubusercontent.com/30750867/109487740-9873cc00-7abf-11eb-8a2d-f199438d52d3.png)

2. import the agent image, I do not know the right format
In other words, how could I replace lines 9 and 10 of the demo code with the self-host agent？
![image](https://user-images.githubusercontent.com/30750867/109488093-0f10c980-7ac0-11eb-82e7-f258e0279054.png)

Thanks

**Comment by egor-bryzgalov** [comment]
Hi @tjinjin95 
You should specify an agent pool name:
```
pool:
  name: 'AgentPoolName'
```
If you'd like to specify an exact agent in the pool you should do the following:
```
pool:
  name: 'AgentPoolName'
  demands:
  - agent.name -equals 'azurePipelineWindows'
```
Is this information helpful for you?

**Comment by tjinjin95** [comment]
@egor-bryzgalov I still cannot specify the agent. The error log is as follows.
code:
![image](https://user-images.githubusercontent.com/30750867/109589333-25149d80-7b45-11eb-8ae3-716d1e4a5b06.png)

first test:
![image](https://user-images.githubusercontent.com/30750867/109589079-be8f7f80-7b44-11eb-8a59-2952afa47b63.png)
second test:
![image](https://user-images.githubusercontent.com/30750867/109589212-f0a0e180-7b44-11eb-9d5b-3d31596ff782.png)

capabilities:
![image](https://user-images.githubusercontent.com/30750867/109588730-327d5800-7b44-11eb-9622-0a3c1f025186.png)

**Comment by tjinjin95** [comment]
Problem solved, could refer to the link: https://github.com/microsoft/azure-pipelines-agent/issues/2996
